### PR TITLE
268 672 new privacy preference on posts

### PR DESCRIPTION
**Pull request by TomBrockington**
add enum for PUBLIC OR PRIVATE
updated controller/posts to accept is private post option
updated user domain to allow update of privacyPref

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -16,6 +16,11 @@ enum Role {
   TEACHER
 }
 
+enum PrivatePosts {
+  PUBLIC
+  PRIVATE
+}
```
Comment (vherus) [inline prisma/schema.prisma:22]: I'm thinking that this enum could potentially be re-used in other models if we change the name of it. What if, down the line, we decide we want to allow profile pages to be public / private? I think a better name for this enum would be `Visibility`

Delete both of the migration files you generated, change the name of this then generate your migrations

File: src/controllers/user.js [hunk 2]
```diff
@@ -127,10 +127,20 @@ export const updateLoggedInUser = async (req, res) => {
 }
 
 export const updateUserById = async (req, res) => {
+  console.log('log')
```
Comment (vherus) [inline src/controllers/user.js:130]: Make sure to remove this log

File: src/controllers/user.js [hunk 3]
```diff
@@ -127,10 +127,20 @@ export const updateLoggedInUser = async (req, res) => {
...
+    email,
+    firstName,
+    lastName,
+    bio,
+    githubUrl,
+    profileImageUrl,
+    privatePosts
+  } = req.body
+
+  console.log('privacyPref', privatePosts)
```
Comment (vherus) [inline src/controllers/user.js:143]: And this one

File: src/controllers/user.js [hunk 4]
```diff
@@ -168,9 +178,12 @@ export const updateUserById = async (req, res) => {
       lastName,
       bio,
       githubUrl,
-      profileImageUrl
+      profileImageUrl,
+      privatePosts
     })
 
+    console.log('updateUser', updateUser)
```
Comment (vherus) [inline src/controllers/user.js:185]: And this one

File: src/domain/user.js [hunk 5]
```diff
@@ -130,8 +137,10 @@ export default class User {
     bio,
     githubUrl,
     profileImageUrl,
-    cohortId
+    cohortId,
+    privatePosts
   }) {
+    console.log('privacyPrefXX', privatePosts)
```
Comment (vherus) [inline src/domain/user.js:143]: And this one

**Review by vherus — CHANGES_REQUESTED** [review]

**Review by vherus — APPROVED** [review]

**Review by miresque — APPROVED** [review]

**Pull request merged**